lib/cursor.js

```diff
@@ -379,9 +378,11 @@ Cursor.prototype.showRecordId = function(value) {
  * Set the cursor snapshot
  * @method
  * @param {object} snapshot The $snapshot operator prevents the cursor from returning a document more than once because an intervening write operation results in a move of the document.
+ * @deprecated as of MongoDB 4.0
  * @return {Cursor}
  */
 Cursor.prototype.snapshot = function(value) {
+  console.warn('Cursor Snapshot is deprecated, and will be removed in a later version');
```

Comment:
basically same comments as `maxScan`: lets create a common function for deprecation (it could even be a function that wraps this function, similar to [depd](https://www.npmjs.com/package/depd#deprecatefunctionfn-message)), and let's warn about deprecated [properties](https://www.npmjs.com/package/depd#deprecatepropertyobj-prop-message) as well.